### PR TITLE
Feat bitvec test

### PR DESCRIPTION
**Pull request by zgfzgf**
optimize bitvec test

### CODE REVIEW

File: common/bitvec/src/lib.rs [hunk 1]
```diff
@@ -31,15 +31,14 @@ const MAX_BUCKETS: usize = 32;
 /// * A position can set more than once -- it remains set after the first time.
 ///
 /// # Examples:
-/// ```ignore
```
Comment (phlip9) [inline common/bitvec/src/lib.rs:34]: Deleting this line will disable to doctest, so let's keep it and change the `ignore` to `rust` so our example doesn't drift : )

File: common/bitvec/src/lib.rs [hunk 2]
```diff
@@ -31,15 +31,14 @@ const MAX_BUCKETS: usize = 32;
...
 /// let mut bv = BitVec::default();
 /// bv.set(2);
 /// bv.set(5);
 /// assert!(bv.is_set(2));
 /// assert!(bv.is_set(5));
 /// assert_eq!(false, bv.is_set(0));
-/// assert_eq!(bv1.count_ones(), 2);
-/// assert_eq!(bv1.last_set_bit(), 3);
+/// assert_eq!(bv.count_ones(), 2);
+/// assert_eq!(bv.last_set_bit(), 5);
```
Comment (phlip9) [inline common/bitvec/src/lib.rs:41]: It look like `last_set_bit` returns an `Option<_>`, so you want

```suggestion
/// assert_eq!(bv.last_set_bit(), Some(5));
```

File: common/bitvec/src/lib.rs [hunk 3]
```diff
@@ -213,13 +212,17 @@ mod test {
         p5.set(10);
         assert!(p5.is_set(10));
         assert_eq!(p5.last_set_bit(), Some(10));
-        assert_eq!(p5.inner, vec![0b0000_0001, 0b0110_0000])
+        assert_eq!(p5.inner, vec![0b0000_0001, 0b0110_0000]);
+
+        let p6 = BitVec { inner: vec![128u8] };
```
Comment (phlip9) [inline common/bitvec/src/lib.rs:217]: let's write this as a binary literal so the test is clearer

```suggestion
        let p6 = BitVec { inner: vec![0b1000_0000] };
```

File: common/bitvec/src/lib.rs [hunk 4]
```diff
@@ -213,13 +212,17 @@ mod test {
...
+        assert_eq!(p6.inner.len(), 1);
+        assert_eq!(p6.last_set_bit(), Some(0));
     }
 
     #[test]
     fn test_empty() {
         let p = BitVec::default();
-        for i in 0..std::u8::MAX {
+        for i in 0..=std::u8::MAX {
```
Comment (phlip9) [inline common/bitvec/src/lib.rs:225]: good catch!

File: common/bitvec/src/lib.rs (hunk 1 above)
Comment (phlip9) [inline common/bitvec/src/lib.rs:34]: I think we also need a `use diem_bitvec::BitVec;` at the start of this doctest

**Review by phlip9 — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just a few minor fixes needed and then we can land this : )

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  M sha2 0.9.3 (direct third-party, crates.io)
    * version upgraded from 0.9.2
    * (unchanged features: default, std)
  M libc 0.2.84 (transitive third-party, crates.io)
    * version upgraded from 0.2.82
    * (unchanged features: default, std)
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  M sha2 0.9.3 (direct third-party, crates.io)
    * version upgraded from 0.9.2
    * (unchanged features: default, std)
  M smallvec 1.6.1 (direct third-party, crates.io)
    * status changed from transitive third-party
    * (unchanged features: [none])
  M libc 0.2.84 (transitive third-party, crates.io)
    * version upgraded from 0.2.82
    * (unchanged features: align, default, std)
  M tokio-util 0.6.3 (transitive third-party, crates.io)
    * version upgraded from 0.6.2
    * (unchanged features: codec, default)
  R async-stream 0.3.0 (transitive third-party, crates.io)
    * (old features: [none])
  R tokio-stream 0.1.2 (transitive third-party, crates.io)
    * (old features: default, time)

host packages:
  R async-stream-impl 0.3.0 (transitive third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A rental 0.5.5 (direct third-party, crates.io)
    * features: default, std
  A stable_deref_trait 1.2.0 (transitive third-party, crates.io)
    * features: alloc, std
  M sha2 0.9.3 (direct third-party, crates.io)
    * version upgraded from 0.9.2
    * (unchanged features: default, std)
  M smallvec 1.6.1 (direct third-party, crates.io)
    * status changed from transitive third-party
    * (unchanged features: [none])
  M tokio-util 0.6.3 (direct third-party, crates.io)
    * version upgraded from 0.6.2
    * (unchanged features: codec, compat, default, futures-io, io)
  M libc 0.2.84 (transitive third-party, crates.io)
    * version upgraded from 0.2.82
    * (unchanged features: align, default, std)
  R async-stream 0.3.0 (transitive third-party, crates.io)
    * (old features: [none])

host packages:
  A rental-impl 0.5.5 (transitive third-party, crates.io)
    * features: [none]
  M syn 1.0.60 (direct third-party, crates.io)
    * added features: fold
    * (unchanged features: clone-impls, default, derive, extra-traits, full, parsing, printing, proc-macro, quote, visit, visit-mut)
  M libc 0.2.84 (transitive third-party, crates.io)
    * version upgraded from 0.2.82
    * (unchanged features: default, std)
  R async-stream-impl 0.3.0 (transitive third-party, crates.io)
    * (old features: [none])
```

**Comment by phlip9** [comment]
Ahh, the reason you're getting all these bot comments is that you included a merge commit in your change set. We explicitly ban merge commits, since we want a linear history (https://github.com/diem/diem/blob/master/CONTRIBUTING.md#responding-to-reviewer-feedback). To fix this, you're going to want to rebase and squash your changes into one commit on upstream/master. Make sure you also drop the merge commit as you're rebasing.

The next time you want to update your branch on top of master, try this instead of `git merge`

```sh
$ git fetch upstream
$ git rebase upstream/master
```

**Review by davidiw — APPROVED** [review]
thanks for the contribution, would probably have just made a single commit for this work in the future, but done is better than perfect :).
/land

**Pull request closed without merging**